### PR TITLE
fix(theming): light text on colored backgrounds should be opaque

### LINKED ISSUE #6236: Text opacity in _theming.scss not in complete alignment
#### Bug, feature request, or proposal:
Proposal

#### What is the expected behavior?
Match Material.io spec for text color usability. Search for "Dark text on light backgrounds" and "White text on dark backgrounds" on this page: https://material.io/guidelines/style/color.html#color-usability

#### What is the use case for this
I am creating a custom theme and want to reuse these colors on my custom controls

#### What is the current behavior?

Currently in _theme.scss

```
$black-87-opacity: rgba(black, 0.87);
$white-87-opacity: rgba(white, 0.87);
$black-12-opacity: rgba(black, 0.12);
$white-12-opacity: rgba(white, 0.12);
$black-6-opacity: rgba(black, 0.06);
$white-6-opacity: rgba(white, 0.06);
```

Proposal

```
// Dark text on light backgrounds
// Primary
$black-87-opacity: rgba(black, 0.87);
// Seconday
$black-54-opacity: rgba(black, 0.54);
// Disabled text, hint text
$black-38-opacity: rgba(black, 0.38);
// Dividers
$black-12-opacity: rgba(black, 0.12);
// NOT IN SPEC...
$black-6-opacity: rgba(black, 0.06);

// Primary
$white-100-opacity: white;
// NOT IN SPEC...
$white-87-opacity: rgba(white, 0.87);
// Seconday
$white-70-opacity: rgba(white, 0.70);
// Disabled text, hint text
$white-50-opacity: rgba(white, 0.50);
// Dividers
$white-12-opacity: rgba(white, 0.12);
// NOT IN SPEC...
$white-6-opacity: rgba(white, 0.06);
```

### PR DESCRIPTION
**Pull request by Maistho**
According to [Material Design Guidelines](https://material.io/guidelines/style/color.html#color-usability), white text on colored backgrounds should be displayed at an opacity of 100%.

I also changed the naming of the variables to [dark/light]-primary-text instead of [black/white]-87-opacity, since it better reflects the names used in the guidelines.

This PR fixes #6236

### CODE REVIEW

**Comment by Maistho** [comment]
I signed it!

**Comment by jelbourn** [comment]
@josephperrott can you take a look to see how this compares to the spec / mdc-web?

**Comment by josephperrott** [comment]
@jelbourn 
These changes match up with the values currently in the [spec](https://material.io/guidelines/style/color.html#color-usability) (under the heading `Text on backgrounds`)

MDC uses the same opacity in their [theming](https://github.com/material-components/material-components-web/blob/master/packages/mdc-theme/_variables.scss#L56)

@Maistho Can you rebase the PR?

**Comment by Maistho** [comment]
@josephperrott I've rebased it!

**Comment by josephperrott** [comment]
@Maistho Looks like there are some build errors now?

**Comment by Maistho** [comment]
@josephperrott Sorry, I had apparently used `$dark-disabled` in some places and `$dark-disabled-text` in others.

Fixed it and rebased on master again.

File: src/lib/core/theming/_palette.scss [hunk 1]
```diff
@@ -6,12 +6,16 @@
...
-$black-12-opacity: rgba(black, 0.12);
-$white-12-opacity: rgba(white, 0.12);
-$black-6-opacity: rgba(black, 0.06);
-$white-6-opacity: rgba(white, 0.06);
+$dark-primary-text: rgba(black, 0.87);
+$dark-secondary-text: rgba(black, 0.54);
+$dark-disabled-text: rgba(black, 0.38);
+$dark-dividers: rgba(black, 0.12);
+$dark-focused: rgba(black, 0.12);
+$light-primary-text: rgba(white, 1);
```
Comment (jelbourn) [inline src/lib/core/theming/_palette.scss]: `rgba(white, 1)` can just be `white`

Comment (Maistho) [reply src/lib/core/theming/_palette.scss]: I felt like it was more clear/explicit that the opacity should be 100% if I typed it out, but I can change it if you prefer it.

Comment (jelbourn) [reply src/lib/core/theming/_palette.scss]: Yeah, I prefer to use the plain value when it is desired

Comment (Maistho) [reply src/lib/core/theming/_palette.scss]: Fixed and rebased

**Comment by jelbourn** [comment]
Caretaker note: this PR will likely affect many screenshot tests and should be synced by itself. Reach out to teams to ensure that everything still looks correct.

**Comment by Maistho** [comment]
Hi, what's the status on this PR? Anything I can do to help speed things up?

**Comment by jelbourn** [comment]
Our merge process is just a bit behind at the moment; see https://github.com/angular/material2/blob/master/CODE_REVIEWS.md#how-prs-are-merged for more context.

This change is particularly involved because it involves updating many screenshot tests.

**Comment by jelbourn** [comment]
@Maistho looks like there is a test failure on CI- can you rebase and see if it goes away?

**Comment by jelbourn** [comment]
@Maistho can you also restore the removed scss variables and mark them as deprecated? Upon running an internal presubmit it turns out quite a large number of apps consume on them.

**Comment by Maistho** [comment]
@jelbourn I've rebased it!

How do I add the deprecation notice? Not sure if you use [sassdoc](http://sassdoc.com/), if I should add it to the commit message, or if there's some other way.

I added sassdoc compatible comments for now.

**Comment by jelbourn** [comment]
Looks good to me

**Comment by andrewseguin** [comment]
@jelbourn Tests pass - do you consider this a patch or minor

**Comment by jelbourn** [comment]
I would consider this acceptable for a patch because it only changes colors

**Comment by andrewseguin** [comment]
@Maistho Thanks for your patience as we try and get this merged in. We're seeing some screenshot diffs get brought up that might point out something incorrect in this PR. 

Can you look at these screenshots and confirm the appearance of the disabled button? Looks like your change made it darker which in my eyes [doesn't match the spec](https://material.io/guidelines/components/buttons.html#buttons-raised-buttons).

**Spec**

xdtf190pzkp

**Master (demo app, button page)**

master

**pr/4721 (demo app, button page)**

new

**Comment by Maistho** [comment]
@andrewseguin Yup, I made a mistake there. Looks like the disabled buttons are using the text color for background. I'll take a look!

**Comment by Maistho** [comment]
@andrewseguin rebased and pushed a fixed version

**Pull request merged**